event-store-plans.md

```diff
@@ -0,0 +1,174 @@
+# Event Store Improvements for V7
```

Comment:
Other ideas that we should include in considerations:
- give users possibility to include their own event type mapping policy (e.g. by strict namespace). That would require changes potentially breaking, as we have event type mapping distributed around the whole codebase, but I think that this is must-have for monoliths,
- potentially adding upcasters,
- open telemetry.